File: src/main/java/com/tom/Hello.java
package com.tom;

public class Hello {
    public static void main(String[] args) {
        /* System.out.println("Hello World"); */

        String s = "abcdef";
        System.out.println(s.charAt(2));
        System.out.println(s.length());
        System.out.println(s.substring(3));
        System.out.println(s.substring(1,4));

        Person p = new Person("Mike",73.8f, 1.694f);
        p.hello();
        p.weight = 55.2f;
        p.height = 1.6f;
        System.out.println(p.bmi());

        int score = 87;
        System.out.println(score<80 || score>90);
        char c = 'A';
        System.out.println(c>'a');

        /*int age = 21;
        Integer age2 = 22;
        char c = '我';
        Character c2 = 'A';
        byte b = 100;
        float weight = 55.2f;
        double height = 163.4;
        boolean adult = true;
        boolean entroll = false;
        String name = "Nancy";*/
    }
}

File: src/main/kotlin/com/tom/Hello.kt
package com.tom

fun main() {
//    println("Hello Kotlin")

    var s:String? = "abcdefg"
    s = null
    println(s?.get(5))
    println(s?.length)
    println(s?.substring(2))
    println(s?.substring(0,5))

    val h = Human(weight = 55.2f, height = 1.6f)
    println(h.bmi())
    h.hello()

    val score = 87
    println(score>85)
    val c = 'A'
    println(c.toInt()>60)

    /*var age = 22
    age = 32
    val weight = 73.2f
    var name1 : String
    name1 = "Mike"*/
}

class Human(var name:String="",var weight:Float, var height:Float){
    init {
        println("test $weight")
    }
//    constructor(name:String, weight: Float, height: Float):this(weight,height)
    fun bmi():Float{
        val bmi=weight/(height*height)
        return bmi
    }
    fun hello(){
        println("Hello Kotlin")
    }
}